content/blog/2019/secure-egress-control-istio/index.md

```diff
@@ -0,0 +1,323 @@
+violate any policy.
+
+Once you direct egress traffic through an egress gateway and apply the additional security mechanisms,
+you can securely monitor the traffic and define security policies on it.
+If the application sends HTTP requests and the egress gateway performs TLS origination, you can monitor HTTP
+information, e.g HTTP methods, headers and URL paths, and you can
+[define policies](/blog/2018/egress-monitoring-access-control) based on the HTTP information. If the application
+performs TLS origination, for TLS traffic you can
+[monitor SNI and the service account](/docs/examples/advanced-gateways/egress_sni_monitoring_and_policies/) of the
+source pod, and define policies based on them.
+
+The following diagram shows Istio Security Architecture, augmented with L3 Firewalls (part of the
+[additional security mechanisms](/docs/examples/advanced-gateways/egress-gateway/#additional-security-considerations)
+provided outside of Istio by the cluster/cloud provider).
+The L3 Firewalls can have a trivial configuration that would allow incoming traffic into Istio ingress gateway pods only and
+outgoing traffic from Istio egress gateway pods only. Note that the Istio proxy of the egress gateway performs
+policy enforcement and reporting in the same way as the sidecar proxies in the application pods.
+
+{{< image width="80%" link="./SecurityArchitectureWithL3Firewalls.svg" caption="Istio Security Architecture with Egress Gateway and L3 Firewalls" >}}
+
+Now let's examine possible attacks and let me show you how the secure egress control in Istio prevents them.
+
+### Possible attacks and their prevention
+
+Consider the following security policy with regard to egress traffic:
+
+1. Application A is allowed to access `*.ibm.com` (all the external services with URL matching `*.ibm.com`,
+  e.g. `www.ibm.com`)
+1. Application B is allowed to access `mongo1.composedb.com`
+1. All the egress traffic must be monitored
+
+Now consider a scenario in which the application A (one of the pods) is compromised. Suppose the attackers have the
+following goals:
+
+1. Application A will try to access `*.ibm.com` unmonitored
+1. Application A will try to access `mongo1.composedb.com`
+
+Note that the application A is allowed to access `*.ibm.com`, so the attacker is able to access it. There is no way
+to prevent such access since there is no way to distinguish, at least initially, between the original and the
+compromised versions of the application A. However, you want to monitor any access to external services to be able to
+detect suspicious traffic, for example by applying anomaly detection tools on logs of the egress traffic.
+The attackers, on the contrary, want to access external services unmonitored, so the attack will not be detected.
+The second goal of the attackers is to access `mongo1.composedb.com`, which is forbidden for the application A. Istio
```

Comment:
for the -> for

Reply:
Fixed in  https://github.com/istio/istio.io/pull/3179/commits/28a9280b8949917fb2a273b3d43e4c56d7fb0c99